docs/BuildOnLinuxOSX.md

```diff
@@ -14,7 +14,7 @@ Firstly, install MLIR (as a part of LLVM-Project):
 ``` bash
 git clone -n https://github.com/llvm/llvm-project.git
 # Check out a specific branch that is known to work with ONNX-MLIR.
-cd llvm-project && git checkout ba8b8a73fcb6b830e63cd8e20c6e13b2a14d69bf && cd ..
+cd llvm-project && git checkout 114ce273d86a091f61446c8778dbf102942c96cf && cd ..
```

Comment:
Changing the llvm commit will not make the tests pass most likely, I suspect you need to make changes to the tests or code somewhere with the green commit.

Also, how was this commit chosen?

Reply:
@sstamenova   Right, its still failing the same.  I picked it to try and narrow down which commit might have caused the backend test failures.

Please let me know of any suggestions for determining the cause of the "running the test CRASHED with signal 6" errors or how to get more details about it.

Reply:
Can you run locally? That would make it much easier to bisect if you decide to do that. 

For the tests, a lot of them have ways of printing extra information, so that would be a good start. Are only the backend tests failing? If the numerical or other tests are failing, they are also easier to diagnose locally.

Reply:
Thanks

I don't see any failures from the numerical or other tests,  just backend tests.  The -jni versions do not fail.

I can recreate in my local docker image using "cmake --build . --config Release --target check-onnx-backend-constant" which gets 360 failures.   And "cmake --build . --config Release --target check-onnx-backend" gets one failure.   I can use TEST_CASE_BY_USER to run just one of the failing tests.  

The tests do not fail in my local macOS native.  They also do not fail on macOS and Windows in the pipeline,  but do fail in the 3 Linux versions in the pipeline.

Reply:
The Windows CI build does not run the backend tests, if I remember correctly. I am not sure about the macOS build.

So locally, on macOS, the backend tests pass? Does that run the same set of tests as the Linux CI builds?

Reply:
The log for the macOS run shows the check-onnx-backend target, but not check-onnx-backend-constant.  The check-onnx-backend tests passed and the number ran for macOS and Linux were the same.  Locally on macOS the backend tests passed and were the same number as Linux.

I did notice that macOS pipeline and local are not the same Python version as Linux.  All 3 Linux runs are Python 3.8.10.  The macOS run is Python 3.10.10.  The local macOS is Python 3.10.9.

I wonder if a change was made that doesn't work well with the backend tests with Python 3.8.10.

Reply:
On Linux, the failure is due to some backend tests are now failing with an assertion failure.
```
# TEST_VERBOSE=true TEST_CASE_BY_USER=test_constant_cpu python3 test/backend/Debug/test.py
Test info:
  temporary results are in dir:./
ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssONNX_HOME=/tmp
Success downloading/saving ./test_constant/test_constant.onnx
cwd: /workdir/onnx-mlir/build
/workdir/onnx-mlir/build/Debug/bin/onnx-mlir -O3 --EmitLib ./test_constant/test_constant.onnx -o=./test_constant/test_constant
IMPORTER FORCE DYNAMIC  None
/workdir/onnx-mlir/build/test/backend/Debug/inference_backend.py:1310: UserWarning: Cannot deduce desired output endianness, using native endianness by default.
  warnings.warn(
/usr/include/c++/9/bits/stl_vector.h:1042: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](std::vector<_Tp, _Alloc>::size_type) [with _Tp = OMTensor*; _Alloc = std::allocator<OMTensor*>; std::vector<_Tp, _Alloc>::reference = OMTensor*&; std::vector<_Tp, _Alloc>::size_type = long unsigned int]: Assertion '__builtin_expect(__n < this->size(), true)' failed.
Aborted (core dumped)
```
A successful run looks like this:
```
# TEST_VERBOSE=true TEST_CASE_BY_USER=test_constant_cpu python3 test/backend/Debug/test.py
Test info:
  temporary results are in dir:./
ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssONNX_HOME=/tmp
Success downloading/saving ./test_constant/test_constant.onnx
cwd: /opt/Downloads/onnx-mlir/build-shared
/root/Downloads/onnx-mlir/build-shared/Debug/bin/onnx-mlir -O3 --EmitLib ./test_constant/test_constant.onnx -o=./test_constant/test_constant
IMPORTER FORCE DYNAMIC  None
/opt/Downloads/onnx-mlir/build-shared/test/backend/Debug/inference_backend.py:1307: UserWarning: Cannot deduce desired output endianness, using native endianness by default.
  warnings.warn(
/usr/local/lib/python3.11/dist-packages/onnx/backend/test/runner/__init__.py:207: DeprecationWarning: `np.object` is a deprecated alias for the builtin `object`. To silence this warning, use `object` by itself. Doing this will not modify any behavior and is safe.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
  if ref_outputs[i].dtype == np.object:
.sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss
----------------------------------------------------------------------
Ran 2492 tests in 15.691s

OK (skipped=2491)
```

Reply:
@gongsu832 Thanks, I was not aware of TEST_VERBOSE, I will pursue this.